### PR TITLE
Feature event type

### LINKED ISSUE #17: Connect booking to backend and show events

### PR DESCRIPTION
**Pull request by michaelbelete**

### CODE REVIEW

File: components/timePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            times.push(hour)
+        }
+    }
+    //sort times so they can arrange accordingly
+    times.sort((a, b) => {
+        return a - b
+    });
+
+    let minute = 0;
+    let index = 1;
```
Comment (simbamentor) [inline components/timePicker.tsx:24]: Foreach already has an index: 
```
array.foreach((value, index) => {...})
```
https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

File: components/timePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                {time}
+            </button >
+        );
+        if (minute < (60 - Number(duration))) {
+            minute = minute + Number(duration);
+        } else {
+            minute = 0
+        }
+        index++;
+    })
```
Comment (simbamentor) [inline components/timePicker.tsx:38]: for now this implementation is fine. we'll probably solve this via the api that returns an object of days with an array of time slots (todo: check how other calendar apps do it)

File: pages/api/booking.ts [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
 
-  const newAttendee = await prisma.attende.create({
+  const newAttendee: Attendee = await prisma.attendee.create({
     data: {
       name: fullName,
       email: email,
     },
   });
 
+  console.log(newAttendee);
```
Comment (simbamentor) [inline pages/api/booking.ts:13]: can be removed

File: prisma/schema.prisma [hunk 4]
```diff
@@ -23,17 +23,27 @@ model User {
...
   user User @relation(fields: [userId], references: [id])
   userId Int 
+  event Event[]
+  @@map("event_type")
+}
+
+model Event {
+  id  Int @default(autoincrement()) @id
+  startDate DateTime @map("start_time")
+  status String //status of the event if the event is approved, rejected, asked to postpone
```
Comment (simbamentor) [inline prisma/schema.prisma:44]: can you make this an enum?

File: pages/api/booking.ts [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
   const newEvent = await prisma.event.create({
     data: {
-      title: title,
-      location: location,
-      start_time: new Date(startTime),
-      end_time: new Date(endTime),
-      user: { connect: { id: userId } },
-      attende: { connect: { id: newAttendee.id } },
+      startDate: new Date(`${date}T${time}:00`),
+      status: 'True',
```
Comment (simbamentor) [inline pages/api/booking.ts:17]: can you make this an enum? is 'True' actually the correct status?

File: pages/api/booking.ts [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
+      eventType: { connect: { id: eventTypeId } },
+      attendee: { connect: { id: newAttendee?.id } },
     },
   });
 
+  console.log(newEvent);
```
Comment (simbamentor) [inline pages/api/booking.ts:23]: can be removed

File: pages/api/eventType.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { getSession } from 'next-auth/client';
+import prisma from '../../lib/prisma';
+
+export default async function handle(req, res) {
+  const { title, duration, location, description } = req.body;
+  const session = await getSession({ req });
```
Comment (simbamentor) [inline pages/api/eventType.ts:6]: there is no actual auth check here which means everyone can create an event type, right?

can you return a 401 if unauthenticated?

File: pages/eventType/[id].tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    const [copySuccess, setCopySuccess] = useState("copy");
+    const [currentDate, setCurrentDate] = useState(new Date())
+    const schedulerData = [];
+
+    var add_minutes = function (dt, minutes) {
+        return new Date(dt.getTime() + minutes * 60000);
+    }
+
+    events.forEach((event) => {
+        console.log(event);
```
Comment (simbamentor) [inline pages/eventType/[id].tsx:68]: console.log

File: pages/eventType/[id].tsx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    let link = "localhost:3000/";
+    if (typeof window === 'object') {
+        link = window.location.origin
+    }
+    const [value, setValue] = useState(`${link}/booking/${eventType?.id}`);
+    const [copySuccess, setCopySuccess] = useState("copy");
+    const [currentDate, setCurrentDate] = useState(new Date())
+    const schedulerData = [];
+
+    var add_minutes = function (dt, minutes) {
```
Comment (simbamentor) [inline pages/eventType/[id].tsx:63]:
```
   const addMinutes = (date, minutes) => new Date(date.getTime() + minutes * 60000);
```

File: pages/eventType/[id].tsx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+        return new Date(dt.getTime() + minutes * 60000);
+    }
+
+    events.forEach((event) => {
```
Comment (simbamentor) [inline pages/eventType/[id].tsx:67]: most `forEach` can be replaced with a simple `map`. in most cases you should use `map`

image

File: pages/index.tsx [hunk 11]
```diff
@@ -4,48 +4,51 @@ import Layout from "../layouts/Landing";
...
+      orderBy: { id: "desc" }
+    });
+    return {
+      props: {
+        user,
+        eventTypes
+      }
+    }
+  } else {
+    // if the user not logged in
```
Comment (simbamentor) [inline pages/index.tsx:41]: any reason why you aren't returning a 401 not authenticated at the top of the function?

File: pages/success.tsx [hunk 12]
```diff
@@ -22,9 +22,9 @@ const Success: React.FC = () => {
             </div>
             <h1 className="mt-3 text-4xl font-bold text-green-700">Success</h1>
             <p className="mt-3 text-gray-800">Your event has been created successfully</p>
-            <Link href="/">
+            {/* <Link href="/">
```
Comment (simbamentor) [inline pages/success.tsx:25]: can be removed?

**Comment by simbamentor** [comment]
good job 🎉

File: pages/api/booking.ts (hunk 5 above)
Comment (michaelbelete) [reply pages/api/booking.ts:17]: no its just a sample im gone make it approve, canceled, disapprove

**Pull request merged**